edb/edgeql/compiler/context.py

```diff
@@ -115,6 +115,34 @@ class PendingCardinality(NamedTuple):
     callbacks: List[PointerCardinalityCallback]
 
 
+class PointerRefCache(
+    Dict[
+        Tuple[s_pointers.PointerLike, s_pointers.PointerDirection],
+        irast.BasePointerRef,
+    ]
+):
+
+    _rcache: Dict[irast.BasePointerRef, s_pointers.PointerLike]
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._rcache = {}
+
+    def __setitem__(
```

Comment:
Since you don't have a `__getitem__` maybe it's better to rename this to `set_ptrcls_for_ref`?

Reply:
This is a dict subclass, `__getitem__` is inherited.